sdk/core/azure-core/azure/core/pipeline/policies/_authentication.py

```diff
@@ -55,10 +56,10 @@ def _enforce_https(request: "PipelineRequest") -> None:
             )
 
     @staticmethod
-    def _update_headers(headers: Dict[str, str], token: str) -> None:
+    def _update_headers(headers: MutableMapping[str, Any], token: str) -> None:
```

Comment:
I don't think we accept headers with Any values, our implementation is quite clear it's `str, str`